core/lib/resources/presence/presence_manager.js

```diff
@@ -39,13 +40,30 @@ PresenceManager.prototype.setup = function() {
   });
 
   // save so you removeListener on destroy
-  this.sentryListener = function(sentry) {
+  this.sentryListener = function (sentry) {
+    if (manager.destroying) return;
```

Comment:
Correct.  Using it there where the sentryListener is being set makes no sense.  I've moved it into chain, where is belongs.  Will also keep the explicit flag.